--- /catkin_ws/src/strands_movebase/strands_navfn/src/strands_navfn_ros.cpp        2019-05-16 05:59:28.217152512 +0000
+++ /catkin_ws/src/strands_movebase/strands_navfn/src/strands_navfn_ros_patched.cpp     2019-05-16 06:35:07.816849093 +0000
@@ -43,7 +43,7 @@
 #include <pcl_conversions/pcl_conversions.h>
 
 //register this planner as a BaseGlobalPlanner plugin
-PLUGINLIB_DECLARE_CLASS(strands_navfn, NavfnROS, strands_navfn::NavfnROS, nav_core::BaseGlobalPlanner)
+PLUGINLIB_EXPORT_CLASS(strands_navfn::NavfnROS, nav_core::BaseGlobalPlanner)
 
 namespace strands_navfn {

