--- /usr/local/lib/python2.7/dist-packages/coverage/cmdline.py  2019-02-28 08:07:46.606178432 +0000
+++ /usr/local/lib/python2.7/dist-packages/coverage/cmdline_sigusr.py   2019-02-28 08:09:35.649866440 +0000
@@ -620,6 +620,11 @@
                 self.coverage.erase()
 
         # Run the script.
+        import signal
+        def handle_flush_signal(signum, stack):
+            self.coverage.save()
+            self.coverage.data.erase()
+        signal.signal(signal.SIGUSR1, handle_flush_signal)
         self.coverage.start()
         code_ran = True
         try:
