--- /usr/local/lib/python2.7/dist-packages/coverage/cmdline.py  2019-01-15 16:15:26.000000000 +0000
+++ /usr/local/lib/python2.7/dist-packages/coverage/cmdline_sigusr.py   2019-01-16 09:53:42.960349210 +0000
@@ -620,6 +620,12 @@
                 self.coverage.erase()
 
         # Run the script.
+        import signal
+        def handle_flush_signal(signum, stack):
+            self.coverage.save()
+            self.coverage._lines = None
+            self.coverage._arcs = None
+        signal.signal(signal.SIGUSR1, handle_flush_signal)
         self.coverage.start()
         code_ran = True
         try:
