--- /catkin_ws/src/robots_stage/launch/stage.launch     2019-03-22 11:11:05.836763976 +0000
+++ /catkin_ws/src/robots_stage/launch/stage_cov.launch 2019-03-22 11:15:22.895988647 +0000
@@ -29,4 +29,8 @@
     <param name="base_watchdog_timeout" value="0.2"/>
   </node>
 <!--node name="rviz" pkg="rviz" type="rviz" args="-d $(find robots_stage)/rviz/rviz_config.rviz"/-->
+  <node pkg="testit_sut" type="testit_service.py" name="testit_sut" output="screen">
+    <param name="node_workspace" value="/catkin_ws"/>
+    <param name="coverage_directories" value="/catkin_ws/build;/root/.ros"/>
+  </node>
 </launch>
